### PR TITLE
Detect JS Engine when reporting errors

### PR DESCRIPTION
**Pull request by jridgewell**
This should save us some time trying to determine what the real browser
is, not what the UserAgent is masquerading as.

### CODE REVIEW

File: test/functional/test-error.js [hunk 1]
```diff
@@ -409,3 +411,40 @@ describes.sandboxed('reportError', {}, () => {
...
   });
 });
+
+describe.only('detectJsEngineFromStack', () => {
+  // Note that these are not true of every case. You can emulate iOS Safari
+  // on Desktop Chrome and break this. These tests are explicitly for
+  // SauceLabs, which runs does not masquerade with UserAgent.
+  describe.configure().ifIos().run('on iOS', () => {
+    it.configure().ifSafari().run('detects safari as safari', () => {
+          expect(detectJsEngineFromStack()).to.equal('Safari');
```
Comment (cramforce) [inline test/functional/test-error.js]: Indent if off in this file.

File: src/error.js [hunk 2]
```diff
@@ -355,3 +367,43 @@ export function detectNonAmpJs(win) {
...
+
+export function detectJsEngineFromStack() {
+  const object = Object.create({
+    // DO NOT rename this property.
+    // DO NOT transform into shorthand method syntax.
+    t: function() {
+      throw new Error('message');
+    }
+  });
+  try {
```
Comment (cramforce) [inline src/error.js]: Why don't we run this on the server side?

File: test/functional/test-error.js (hunk 1 above)
Comment (jridgewell) [inline test/functional/test-error.js]: Fix.

File: src/error.js (hunk 2 above)
Comment (jridgewell) [inline src/error.js]: Firefox and IE return different results based on how the prototype is constructed.

```js
function Klass() {}
Klass.prototype.t

// Different than
proto = { t };
obj = Object.create(proto);
```

It'd have to do much more complicated testing if I don't know for certain how the chain is constructed.

File: test/functional/test-error.js [hunk 3]
```diff
@@ -409,3 +410,40 @@ describes.sandboxed('reportError', {}, () => {
     }).to.throw(/_reported_ Error reported incorrectly/);
   });
 });
+
+describe.only('detectJsEngineFromStack', () => {
```
Comment (cramforce) [inline test/functional/test-error.js]: Can you update the integration test glob to include this (so the tests actually run on SC)?

File: src/error.js (hunk 2 above)
Comment (cramforce) [inline src/error.js]: K. Please update errortracker.go to append this to the UA string (maybe wrapped in [] or something like that).

File: test/functional/test-error.js (hunk 3 above)
Comment (jridgewell) [inline test/functional/test-error.js]: Done.

**Pull request merged**